test/EndToEndTests/Tests/Client/AsynchronousTests/AsynchronousQueryTests.cs

```diff
@@ -788,6 +789,24 @@ public void Linq_ProjectPropertiesFromEntityandExpandedEntity()
             this.EnqueueTestComplete();
         }
 
+        [Fact]
+        public async Task Linq_ProjectPropertiesFromEntityWithConditionalNullCheckOnExpandedEntity()

```

Comment:
Would this test be the one that fails without your other changes? The LINQ query wasn't generating an expression with the materializer?

Reply:
Yes, it was failing without my changes. It was throwing an `ArgumentException` because the `ProjectPlanCompiler` was generating a dynamic call to `ProjectionCheckValueForPathIsNull` without passing the required `materializerContext` argument.